apps/fares/lib/min_max_fare.ex

```diff
@@ -17,22 +15,32 @@ defmodule Site.BaseFare do
 
   @default_trip %Trip{name: "", id: ""}
 
-  @spec base_fare(
+  @spec lowest_fare(
           Route.t() | map,
           Trip.t() | map,
           Stops.Stop.id_t(),
           Stops.Stop.id_t(),
           (Keyword.t() -> [Fare.t()])
         ) ::
           String.t() | nil
-  def base_fare(route, trip, origin_id, destination_id, fare_fn \\ &Repo.all/1)
-  def base_fare(nil, _, _, _, _), do: nil
+  def lowest_fare(route, trip, origin_id, destination_id, fare_fn \\ &Repo.all/1)
+  def lowest_fare(nil, _, _, _, _), do: nil
 
-  def base_fare(route, nil, origin_id, destination_id, fare_fn) do
-    base_fare(route, @default_trip, origin_id, destination_id, fare_fn)
+  def lowest_fare(route, nil, origin_id, destination_id, fare_fn) do
+    lowest_fare(route, @default_trip, origin_id, destination_id, fare_fn)
   end
 
-  def base_fare(route, trip, origin_id, destination_id, fare_fn) do
+  def lowest_fare(route, trip, origin_id, destination_id, fare_fn) do
+    get_fares(route, trip, origin_id, destination_id, fare_fn)
+    |> Enum.min_by(& &1.cents, fn -> nil end)
```

Comment:
```suggestion
    route
    |> get_fares(trip, origin_id, destination_id, fare_fn)
    |> Enum.min_by(& &1.cents, fn -> nil end)
```

This is a real minor style thing, but in Elixir there's a general convention of starting pipe chains with data, not with a function call. In practice that generally just means pulling out the first argument to your first function in the chain.

Reply:
Yes, after looking at the code it seemed like this could be a simple enough solution. I didn't add the tests because I wasn't sure this was a good approach. I'll do that now and will also address your change requests.